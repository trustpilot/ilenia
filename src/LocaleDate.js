import withTranslations from './withTranslations';
import PropTypes from 'prop-types';

const LocaleDate = ({ date, locale, format }) => {
  const dateWrapper = new Date(date);

  if (isNaN(dateWrapper)) {
    console.error('Invalid date');
    return null;
  }

  try {
    return dateWrapper.toLocaleDateString(locale, format);
  } catch (err) {
    return dateWrapper.toLocaleDateString('en-US', format);
  }
};

LocaleDate.propTypes = {
  date: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.instanceOf(Date)]),
  format: PropTypes.shape({
    weekday: PropTypes.oneOf(['narrow', 'short', 'long']),
    era: PropTypes.oneOf(['narrow', 'short', 'long']),
    year: PropTypes.oneOf(['numeric', '2-digit']),
    month: PropTypes.oneOf(['numeric', '2-digit', 'narrow', 'short', 'long']),
    day: PropTypes.oneOf(['numeric', '2-digit']),
    hour: PropTypes.oneOf(['numeric', '2-digit']),
    minute: PropTypes.oneOf(['numeric', '2-digit']),
    second: PropTypes.oneOf(['numeric', '2-digit']),
    timeZoneName: PropTypes.oneOf(['short', 'long']),
  }),
};

export default withTranslations(LocaleDate);
