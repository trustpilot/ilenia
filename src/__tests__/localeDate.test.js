import React from 'react';
import renderer from 'react-test-renderer';

import LocalizationProvider from '../LocalizationProvider';
import LocaleDate from '../LocaleDate';

test('Renders a localized date from a number', () => {
  const component = renderer.create(
    <LocalizationProvider locale="en-US" translations={{}}>
      <LocaleDate date={0} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a localized date from a string', () => {
  const component = renderer.create(
    <LocalizationProvider locale="en-US" translations={{}}>
      <LocaleDate date={'01 Jan 1970 00:00:00 GMT'} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a localized date from a date', () => {
  const component = renderer.create(
    <LocalizationProvider locale="en-US" translations={{}}>
      <LocaleDate date={new Date(0)} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders null if the date is invalid', () => {
  console.error = jest.fn();

  const component = renderer.create(
    <LocalizationProvider locale="en-US" translations={{}}>
      <LocaleDate date={'abcdefghijklmnopqrstuvwxyz'} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(console.error).toHaveBeenCalled();
});

test('Renders a localized date with an incorrect locale', () => {
  const component = renderer.create(
    <LocalizationProvider locale="abcdefghijklmnopqrstuvwxyz" translations={{}}>
      <LocaleDate date={0} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('REnders a localized date from UTC date string', () => {
  const component = renderer.create(
    <LocalizationProvider locale="abcdefghijklmnopqrstuvwxyz" translations={{}}>
      <LocaleDate date="2018-07-10T12:38:37.713Z" />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
