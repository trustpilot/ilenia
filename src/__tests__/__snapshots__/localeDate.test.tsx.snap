// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a localized date from UTC date string 1`] = `7/10/2018`;

exports[`Renders a localized date from a date 1`] = `1/1/1970`;

exports[`Renders a localized date from a number 1`] = `1/1/1970`;

exports[`Renders a localized date from a string 1`] = `1/1/1970`;

exports[`Renders a localized date with an incorrect locale 1`] = `1/1/1970`;

exports[`Renders a localized date with specific format 1`] = `Jul 10, 2018`;

exports[`Renders null if the date is invalid 1`] = `null`;
